web/src/components/Beneficiaries/DropdownBeneficiary.tsx

```diff
@@ -0,0 +1,118 @@
+import classnames from 'classnames';
+import React, { useEffect, useState, useMemo, useCallback } from 'react';
+import { Dropdown } from 'react-bootstrap';
+import { ChevronIcon } from '../../assets/images/ChevronIcon';
+import { convertToString } from '../../helpers';
+import { BeneficiariesBlockchainItemProps } from './BeneficiariesCrypto/BeneficiariesBlockchainItem';
+
+type DropdownElem = number | string | React.ReactNode;
+
+export interface DropdownBeneficiaryProps {
+    /**
+     * List of options
+     */
+    list: DropdownElem[] | any;
+    /**
+     * Selection callback function
+     * @default empty
+     */
+    onSelect?: (index: number) => void;
+    /**
+     *  By default class name 'cr-dropwdown'
+     *  This property gives an additional class name
+     *  @default empty
+     */
+    className?: '',
+    /**
+     * Value for placeholder of Dropdown components
+     * @default empty
+     */
+    placeholder?: string;
+    /**
+     * Value for disabling contentEditable property
+     * @default false
+     */
+    disableContentEditable?: boolean;
+    /**
+     * Value for clear selected item
+     * @default false
+     */
+    clear?: boolean;
+    selectedValue?: BeneficiariesBlockchainItemProps;
+}
+
+const defaultSelected = {
+    blockchainKey: '',
+    protocol: '',
+    name: '',
+    id: '',
+    fee: '',
+    minWithdraw: '',
+}
+
+/**
+ *  Cryptobase Dropdown that overrides default dropdown with list of options.
+ */
+
+export const DropdownBeneficiary = (props: DropdownBeneficiaryProps) => {
+    const [selected, setSelected] = useState<BeneficiariesBlockchainItemProps | undefined>(defaultSelected);
+
+    const { list, className, placeholder, clear, onSelect } = props;
+    const defaultPlaceholder = list[0];
+
+    const cx = useMemo(() => classnames('cr-dropdown', className, {
+        'cr-dropdown--default': defaultSelected.protocol === placeholder,
+    }), [selected, placeholder, className]);
+
+    useEffect(() => {
+        if (clear !== false) {
+            setSelected(placeholder || convertToString(defaultPlaceholder));
+        }
+    }, [placeholder, defaultPlaceholder, clear]);
+
+    useEffect(() => {
+        if (typeof props.selectedValue !== 'undefined' && props.selectedValue.id !== '') {

```

Comment:
props.selectedValue.id is equal to `props.selectedValue.id !== ''`
since `id` is string you can use this check or convert props.selectedValue.id to string

Reply:
it's optional param that's why it can have a type undefined